numpyro/distributions/distribution.py

```diff
@@ -22,11 +22,48 @@
 # ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 # POSSIBILITY OF SUCH DAMAGE.
 
+from contextlib import contextmanager
+import warnings
+
 import jax.numpy as np
 
 from numpyro.distributions.constraints import is_dependent
 from numpyro.distributions.transforms import Transform
-from numpyro.distributions.util import lazy_property, sum_rightmost
+from numpyro.distributions.util import lazy_property, sum_rightmost, validate_sample
+from numpyro.util import not_jax_tracer
+
+_VALIDATION_ENABLED = False
+
+
+def enable_validation(is_validate=True):
```

Comment:
This can be a more general utility that can be used for validation outside the context of distributions. We should move this to `util` and import it in `numpyro.__init__` instead.

Reply:
I exposed this to numpyro namespace. The reason for this to be here is to avoid circular import: these utils require access to Distribution class.

Reply:
hmm..yeah, I think lets keep it here, in that case. If we have more validation checks we can do provide a utility method to do it.